### DISCUSSION TITLE
Does the caller function information work for c with clangd server?

### DISCUSSION THREAD

**Discussion by cpcopy** [ray-x/navigator.lua#303] [Q&A]
From the demo, I can see the caller function is shown for references. Is it also supported for c projects? I tried the plugin, but didn't see such information. Does it need specific configuration?

**Comment by ray-x** [comment]
The caller function requires treesitter c parser to work correctly. 
![image](https://github.com/ray-x/navigator.lua/assets/1681295/afb913f5-7c53-492a-8256-f11f8ac6e7cf)

**Reply by cpcopy** [reply]
Thanks. I have nvim-treesitter installed. Tried a simple c code. It works. But it doesn't work on my working project. Not sure why.
![image](https://github.com/ray-x/navigator.lua/assets/80554779/bfe579f4-d859-4117-afb6-9eaa6b125eb2)
![image](https://github.com/ray-x/navigator.lua/assets/80554779/03138b8e-1667-491e-843a-fde6162b3d51)

**Comment by cpcopy** [comment]
I have roughly found out the reason. It doesn't work when I open the project from the dashboard. But it works when I open nvim under the project root folder. But don't know the exact reason.

**Reply by ray-x** [reply]
What dashboard plugin are you using? I do not use dashboard. Also as most of lsp server depend on rood folder to work correctly. 
If you open file from dashboard, pls make sure the root folder is setup correctly.

**Reply by cpcopy** [reply]
It is https://github.com/nvimdev/dashboard-nvim. I select the project from the board and it will switch pwd to the project. Is it possible navigator loaded first and doesn't get the changed pwd?
